### PR TITLE
Change climate code, Fix some weird code, Make slightly smoother edges at the cost of efficiency

### PR DESCRIPTION
**Pull request by valoeghese**
Please make my new chunk gen code use a noise cache so it isn't slow thx

### CODE REVIEW

**Comment by jaskarth** [comment]
i will unslow ur code

File: src/main/java/supercoder79/simplexterrain/world/gen/SimplexBiomeSource.java [hunk 1]
```diff
@@ -41,40 +46,17 @@ public void setHeightmap(Heightmap heightmap) {
...
-			else return Biomes.OCEAN;
-		}
-		if (height < SimplexTerrain.CONFIG.lowlandStartHeight) return Biomes.BEACH;
-		if (height < SimplexTerrain.CONFIG.midlandStartHeight)  return lowlandsSampler.sample(x, z);
-		if (height < SimplexTerrain.CONFIG.highlandStartHeight) return midlandsSampler.sample(x, z);
-		if (height < SimplexTerrain.CONFIG.toplandStartHeight) return highlandsSampler.sample(x, z);
-		return toplandsSampler.sample(x, z);
+		if (height < SimplexTerrain.CONFIG.seaLevel / 2) return this.deepOceanSampler.sample(x, z);
+		if (height < SimplexTerrain.CONFIG.seaLevel - 7) return this.oceanSampler.sample(x, z);
+		if (height < SimplexTerrain.CONFIG.seaLevel - 2) return Biomes.OCEAN;
```
Comment (jaskarth) [inline src/main/java/supercoder79/simplexterrain/world/gen/SimplexBiomeSource.java]: why does it return ocean here

**Review by jaskarth — CHANGES_REQUESTED** [review]

File: src/main/java/supercoder79/simplexterrain/world/gen/SimplexBiomeSource.java (hunk 1 above)
Comment (valoeghese) [reply src/main/java/supercoder79/simplexterrain/world/gen/SimplexBiomeSource.java]: so that water colour on biome edges don't look weird

Comment (valoeghese) [reply src/main/java/supercoder79/simplexterrain/world/gen/SimplexBiomeSource.java]: suddenly getting coloured a few blocks out

Comment (valoeghese) [reply src/main/java/supercoder79/simplexterrain/world/gen/SimplexBiomeSource.java]: you can tweak the cutoff values /shrug

Comment (jaskarth) [reply src/main/java/supercoder79/simplexterrain/world/gen/SimplexBiomeSource.java]: the beaches already do that right

**Comment by valoeghese** [comment]
Damn it there are merge conflicts

**Comment by jaskarth** [comment]
it's ok leave it to me

**Review by jaskarth — APPROVED** [review]

**Pull request merged**